### ISSUE TITLE
Helm 3 does not see to respect the --kubeconfig flag

### ISSUE THREAD

**Issue by limscoder** [helm/helm#5950]
Output of `helm version`:

version.BuildInfo{Version:"v3.0.0-alpha.1", GitCommit:"b9a54967f838723fe241172a6b94d18caf8bcdca", GitTreeState:"clean"}

Output of `kubectl version`:

Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:02:58Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:02:58Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}

V3 does not seem to respect the --kubeconfig flag.

Example:

```
[root@host kubernetes]# helm list
NAME        	NAMESPACE	REVISION	UPDATED                                	STATUS  	CHART
my-chart	default  	1       	2019-06-27 17:20:22.438150461 +0000 UTC	deployed	my-chart-0.1.0
[root@host kubernetes]# mv ~/.kube/config ~/.kube/config.bak
[root@host kubernetes]# helm list
NAME	NAMESPACE	REVISION	UPDATED	STATUS	CHART
[root@host kubernetes]# helm --kubeconfig=~/.kube/config.bak list
NAME	NAMESPACE	REVISION	UPDATED	STATUS	CHART
[root@host kubernetes]#
```

**Comment by limscoder** [comment]
it appears that setting KUBECONFIG works as expected.

**Comment by hickeyma** [comment]
Thanks @limscoder. This seems to be a bug alright.

I tested it as follows:

```console
$ cat ~/.kube/config
apiVersion: v1
clusters:
- cluster:
    server: http://127.0.0.1:32768
  name: dind
- cluster:
    server: ""
  name: test
contexts:
- context:
    cluster: dind
    namespace: default
    user: ""
  name: dind
- context:
    cluster: dind
    namespace: test
    user: ""
  name: test
current-context: test
kind: Config
preferences: {}
users: []

$ helm ls
NAME	NAMESPACE	REVISION	UPDATED	STATUS	CHART

$ cat ~/.kube/config.bak 
apiVersion: v1
clusters:
- cluster:
    insecure-skip-tls-verify: true
    server: http://127.0.0.1:32768
  name: dind
contexts:
- context:
    cluster: dind
    user: ""
  name: dind
current-context: dind
kind: Config
preferences: {}

$ helm --kubeconfig=~/.kube/config.bak list
NAME	NAMESPACE	REVISION	UPDATED	STATUS	CHART

$ mv ~/.kube/config.bak ~/.kube/config

$ helm ls
NAME   	NAMESPACE	REVISION	UPDATED                                	STATUS  	CHART        
mychart	default  	1       	2019-06-28 11:09:43.763417029 +0100 IST	deployed	mychart-0.1.0
```

**Comment by jan25** [comment]
Thanks for raising this issue! Looks like we load flags first and then overwrite them with environment variables, its here in [root.go](https://github.com/helm/helm/blob/dev-v3/cmd/helm/root.go#L120)
@hickeyma would you know if that is desired behaviour? I suppose they should be ordered other way around to fix this issue.

**Comment by jan25** [comment]
Thanks for raising this issue! Looks like this is a problem with all [EnvSettings](https://github.com/helm/helm/blob/dev-v3/pkg/cli/environment.go#L53) flags.

The apparent reason is we initialise the rootCmd in root.go after setting up [kube client](https://github.com/helm/helm/blob/dev-v3/cmd/helm/helm.go#L68) in helm.go. And the flags are only [parsed](https://github.com/helm/helm/blob/dev-v3/cmd/helm/root.go#L120) in rootCmd, which are not considered to update to kube client

**Comment by OJFord** [comment]
Nor perhaps any flag?

At least `helm install --namespace` is ignored too, so everything is deployed to `default`.

**Comment by thomastaylor312** [comment]
Closed by #6025
